src/ctrl/menu/darwin.rs

```diff
@@ -0,0 +1,130 @@
+use crate::ctrl::Controller;
+use crate::ctrl::Event;
+use crate::ui::View;
+use crate::utils::Aro;
+use crate::Model;
+
+use muda::Menu;
+use muda::MenuItemBuilder;
+use muda::SubmenuBuilder;
+use muda::AboutMetadataBuilder;
+use muda::AboutMetadata;
+use muda::Submenu;
+use muda::PredefinedMenuItem;
+
+use std::sync::mpsc::Sender;
+
+pub struct DarwinMenu;
+
+impl Controller for DarwinMenu {
+    fn setup(_: Aro<Model>, _: &View, tx: Sender<Event>) {
+        let app = SubmenuBuilder::new()
+            .enabled(true)
+            .text("&Kira")
+            .items(&[
+                &PredefinedMenuItem::separator(),
+                &PredefinedMenuItem::about(None, Some(about())),
+                &PredefinedMenuItem::separator(),
+                &PredefinedMenuItem::hide(None),
+                &PredefinedMenuItem::hide_others(None),
+                &PredefinedMenuItem::show_all(None),
+                &PredefinedMenuItem::services(None),
+                &PredefinedMenuItem::separator(),
+                &PredefinedMenuItem::quit(None),
+            ])
+            .build()
+            .expect("Failed to build App submenu");
+
+        let file = SubmenuBuilder::new()
+            .enabled(true)
+            .text("&File")
+            .items(&[
+                &MenuItemBuilder::new()
+                    .enabled(true)
+                    .text("Open Project...")
+                    .id("open-file".into())
+                    .acccelerator(Some("CMD+O"))
+                    .unwrap()
+                    .build(),
+                &MenuItemBuilder::new()
+                    .enabled(true)
+                    .text("Save")
+                    .id("save".into())
+                    .acccelerator(Some("CMD+S"))
+                    .unwrap()
+                    .build(),
+                &MenuItemBuilder::new()
+                    .enabled(true)
+                    .text("Save as...")
+                    .id("save-as".into())
+                    .acccelerator(Some("CMD+SHIFT+S"))
+                    .unwrap()
+                    .build(),
+                &PredefinedMenuItem::separator(),
+                &MenuItemBuilder::new()
+                    .enabled(true)
+                    .text("Open Command Palette...")
+                    .id("command-palette".into())
+                    .acccelerator(Some("CMD+K"))
+                    .unwrap()
+                    .build(),
+                &MenuItemBuilder::new()
+                    .enabled(true)
+                    .text("Render Current Segment")
+                    .id("render".into())
+                    .acccelerator(Some("CMD+R"))
+                    .unwrap()
+                    .build(),
+            ])
+            .build()
+            .unwrap();
+
+        let menu_bar = Menu::new();
+
+        menu_bar.append_items(&[&app, &file]).unwrap();
+        menu_bar.init_for_nsapp();
```

Comment:
we need a gtk window for the linux version of this, but in general, this entire thing should *not* be in a platform specific module if we're going to use it for multiple platform, this isn't a darwin menu, it's just the menu.